### DISCUSSION TITLE
time limit exceeded `test.jpg' @ fatal/cache.c/GetImagePixelCache/1867

### DISCUSSION THREAD

**Discussion by TobiasKrais** [ImageMagick/ImageMagick6#345] [Q&A]
Hi there,

you're my last hope for solution. I use the following php script:

```php
$start = microtime(true);

try {
    $imagick = new \Imagick('test.jpg');
} catch (\Exception $e) {
    echo $e->getMessage() .'<br>';
}

// calculate execution time
echo 'Execution time: '. microtime(true) - $start;
```

The script output is:
```
time limit exceeded `attachments/2023_05/553721_3_2023-02-10_13.14.24.jpg' @ fatal/cache.c/GetImagePixelCache/1867
Execution time: 0.0014991760253906
```

After loading the script 3 or 4 times the failure is gone. But after waiting some minutes, the failure occurs again. Do you have any idea how I can find out what causes the exception?

System info:

- Ubuntu 24.04 with plesk-php83-imagick (package version: 3.7.0-ubuntu.24.04.240926.1652, imagick version: 6.9.12-98 Q16 x86_64 18038)
- Apache and nginx
- PHP 8.3

**Comment by dlemstra** [comment]
It looks like your installation has a time limit in the `policy.xml` configuration?

**Reply by TobiasKrais** [reply]
Is this the correct line?

`<policy domain="resource" name="time" value="960"/>`

Looks OK for me.

**Reply by dlemstra** [reply]
Might be a bug in an old version, what happens if you upgrade to the most recent version of ImageMagick?

**Reply by TobiasKrais** [reply]
According to my hoster, it is not possible to upgrade the ImageMagick to the most recent version. Do you mean version 7 or latest version 6?

**Reply by dlemstra** [reply]
The latest version of either.

**Reply by TobiasKrais** [reply]
I just upgraded my local computer to Ubuntu 24.10 with imagemagick 8:6.9.13.12+dfsg1-1. The particular error is gone. But now `$imagick->getImageBlob()` returns an empty string when image is image/heic.

Thus the original issue seems to be an already fixed bug.

**Reply by dlemstra** [reply]
I suspect that you don't have support for writing heic files in your local installation.

**Reply by TobiasKrais** [reply]
`libheif` is installed on the system. Shouldn't that be sufficient?
